--- a/WebKit.pri
+++ b/WebKit.pri
@@ -52,6 +52,7 @@ INCLUDEPATH += $$OUTPUT_DIR/include/QtWe
 
 CONFIG -= warn_on
 *-g++*:QMAKE_CXXFLAGS += -Wall -Wreturn-type -fno-strict-aliasing -Wcast-align -Wchar-subscripts -Wformat-security -Wreturn-type -Wno-unused-parameter -Wno-sign-compare -Wno-switch -Wno-switch-enum -Wundef -Wmissing-noreturn -Winit-self
+*-g++*:QMAKE_LFLAGS += -Wl,--as-needed
 *-g++*:QMAKE_LFLAGS += -Wl,--version-script,$$BASE_DIR/symbols.filter
 
 # Enable GNU compiler extensions to the ARM compiler for all Qt ports using RVCT
