Description: Add support for s390/s390x architectures
Author: Dan Horák <dan@danny.cz>
Origin: vendor (http://trac.webkit.org/changeset?old_path=%2Ftrunk&old=71843+&new_path=%2Ftrunk&new=71844+)
Forwarded: no
Last-Update: 2011-08-06

--- a/Source/JavaScriptCore/wtf/Platform.h
+++ b/Source/JavaScriptCore/wtf/Platform.h
@@ -186,6 +186,18 @@
 #define WTF_CPU_SPARC 1
 #endif

+/* CPU(S390X) - S390 64-bit */
+#if defined(__s390x__)
+#define WTF_CPU_S390X 1
+#define WTF_CPU_BIG_ENDIAN 1
+#endif
+
+/* CPU(S390) - S390 32-bit */
+#if defined(__s390__)
+#define WTF_CPU_S390 1
+#define WTF_CPU_BIG_ENDIAN 1
+#endif
+
 /* CPU(X86) - i386 / x86 32-bit */
 #if   defined(__i386__) \
     || defined(i386)     \
@@ -917,6 +929,7 @@
 #if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS))) \
     || (CPU(IA64) && !CPU(IA64_32)) \
     || CPU(ALPHA) \
+    || CPU(S390X) \
     || CPU(SPARC64)
 #define WTF_USE_JSVALUE64 1
 #elif CPU(ARM_TRADITIONAL) || CPU(PPC64) || CPU(MIPS)

